.github/composite/exp2/action.yml

```diff
@@ -0,0 +1,24 @@
+name: Run experiment 2
+
+inputs:
+  repositoryUrl:
+    required: true
+    type: string
+  branch:
+    required: true
+    type: string
+  task:
+    required: true
+    type: string
+  gradleEnterpriseUrl:
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Run Gradle Experiment 2
+      run: |
+        cd gradle-enterprise-gradle-build-validation
+        ./02-validate-local-build-caching-same-location.sh -r ${{ inputs.repositoryUrl }} -b ${{ inputs.branch }} -t ${{ inputs.task }} -s ${{ inputs.gradleEnterpriseUrl }}
+      shell: bash
```

Comment:
It might be useful to make the shell configurable. For example, this would allow us to use these actions in our cross-platform tests (where we sometimes have to use a shell that is different from bash).

Reply:
I would have been happy to do so but the templated syntax does not apply for `shell`.

What is the exact use case @jthurne as the so called `bash` _shell_ supports all platforms (_Windows_ included):
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell

> The default shell on non-Windows platforms with a fallback to sh. When specifying a bash shell on Windows, the bash shell included with Git for Windows is used.

Reply:
> I would have been happy to do so but the templated syntax does not apply for `shell`.

Oh yeah, that's right. I ran into that with the cross-version tests as well.

> What is the exact use case @jthurne as the so called bash shell supports all platforms (Windows included):

The shell setting of "bash" on Windows uses the version of bash that comes with Git on Windows, but it isn't a complete bash implementation. Most people who use Windows instead use a full Linux version of bash under the Windows Linux Subsystem (WSL).  It isn't possible to redefine the `bash` in the GitHub Actions to refer to the WSL bash instead. So instead, we need to use `wsl-bash` for our Windows tests (see [the Setup WSL GH Action](https://github.com/marketplace/actions/setup-wsl)).

Since we cannot make it configurable, it may be better not to set a `shell` at all. On most platforms, `bash` is the default shell anyway. On other platforms, like Windows, not setting a shell would allow the user to set the default shell to something else (like [what is done in our cross platform tests](https://github.com/gradle/gradle-enterprise-build-validation-scripts/blob/main/.github/workflows/cross-platform-testing.yml#L32)).

Reply:
The `shell` instruction is mandatory in a composite action, this being said, this should be ok to have `bash` for the steps executed within the composite action and another `shell` for other steps. wdyt?

Reply:
> The shell instruction is mandatory in a composite action

That's a bummer.

> this should be ok to have bash for the steps executed within the composite action and another shell for other steps

For our cross platform tests, it's important that the actual validation scripts are run in the WSL bash rather than Git Bash. If we cannot rely on the default shell configured for the workflow, then that just means we won't be able to use these composite actions with our cross platform tests. 

But that's just one GitHub workflow. It still means they can be used across the open source projects, so there's still value.

Reply:
Yep, just updated my comment meanwhile...
I think we could try to find a way with some conditional steps, let me come with a proposal.